pkg/gadgets/run/tracer/tracer.go

```diff
@@ -1052,6 +1081,30 @@ func (t *Tracer) runToppers(gadgetCtx gadgets.GadgetContext) {
 	}
 }
 
+func (t *Tracer) collectProfiler(gadgetCtx gadgets.GadgetContext) error {
+	cb := t.processEventFunc(gadgetCtx)
+
+	events := []*types.Event{}
+
+	key := make([]byte, t.histMap.KeySize(), t.histMap.KeySize())
+	value := make([]byte, t.histMap.ValueSize(), t.histMap.ValueSize())
+
+	it := t.histMap.Iterate()
+	for it.Next(&key, &value) {
+		event := cb(value)
+		events = append(events, event)
+	}
+
+	err := it.Err()
+	if err != nil {
+		return err
```

Comment:
nit: Provide more context

```suggestion
		return fmt.Errorf("iterating over profiler map: %w", err)
```